drivers/digital-io/max22200/iio_max22200.c

```diff
@@ -0,0 +1,1212 @@
+ * @param buf - Buffer to be filled with requested data.
+ * @param len - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv - Private descriptor
+ * @return 0 in case of success, error code otherwise
+ */
+static int max22200_iio_write_channel_freq(void *dev, char *buf, uint32_t len,
+		const struct iio_ch_info *channel,
+		intptr_t priv)
+{
+	struct max22200_iio_desc *desc = dev;
+	enum max22200_ch_freq ch_freq;
+	size_t i;
+
+	for (i = 0; i < NO_OS_ARRAY_SIZE(max22200_channel_freq_avail); i++) {
+		if (!strcmp(buf, max22200_channel_freq_avail[i])) {
+			ch_freq = i;
+
+			return max22200_set_ch_freq(desc->max22200_desc, channel->address, ch_freq);
+		}
+	}
+
+	return -EINVAL;
+}
+
+/**
+ * @brief Write the channel_enable attribute for a specific channel.
+ * @param dev - The iio device structure.
+ * @param buf - Buffer to be filled with requested data.
+ * @param len - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv - Private descriptor
+ * @return 0 in case of success, error code otherwise
+ */
+static int max22200_iio_write_enable(void *dev, char *buf, uint32_t len,
+				     const struct iio_ch_info *channel,
+				     intptr_t priv)
+{
+	struct max22200_iio_desc *desc = dev;
+	uint32_t val;
+
+	iio_parse_value(buf, IIO_VAL_INT, (int32_t *)&val, NULL);
+
+	return max22200_set_ch_enable(desc->max22200_desc, channel->address,
+				      priv, val);
+}
+
+/**
+ * @brief Write the fault_mask debug attribute for MAX22200 IIO device.
+ * @param dev - The iio device structure.
+ * @param buf - Buffer to be filled with requested data.
+ * @param len - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv - Private descriptor
+ * @return 0 in case of success, error code otherwise
+ */
+static int max22200_iio_write_fault_mask(void *dev, char *buf, uint32_t len,
+		const struct iio_ch_info *channel,
+		intptr_t priv)
+{
+	struct max22200_iio_desc *desc = dev;
+	uint32_t val;
+
+	iio_parse_value(buf, IIO_VAL_INT, (int32_t *)&val, NULL);
+
+	if (val > 1)
+		val = 1;
+
+	return max22200_fault_mask_set(desc->max22200_desc, priv, val);
+}
+
+/**
+ * @brief Write the fault_mask debug attribute for MAX22200 IIO device.
+ * @param dev - The iio device structure.
+ * @param buf - Buffer to be filled with requested data.
+ * @param len - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv - Private descriptor
+ * @return 0 in case of success, error code otherwise
+ */
+static int max22200_iio_write_trig_state(void *dev, char *buf, uint32_t len,
+		const struct iio_ch_info *channel,
+		intptr_t priv)
+{
+	struct max22200_iio_desc *desc = dev;
+	uint32_t val;
+
+	iio_parse_value(buf, IIO_VAL_INT, (int32_t *)&val, NULL);
+
+	if (val > 1)
+		val = 1;
+
+	return max22200_set_trig_state(desc->max22200_desc, val);
```

Comment:
so here the function takes an enumeration as parameter and you are giving a uint32_t value. I would replace this code with something like:
```
return max22200_set_trig_state(desc->max22200_desc, val ? MAX22200_TRIG_ON : MAX22200_TRIG_OFF);
```
which shows better what is the intended purpose and what val represents
see in all APIs where applicable

Reply:
sure.